src/toil/test/cwl/cwlTest.py

```diff
@@ -707,6 +721,53 @@ def test_kubernetes_cwl_conformance_with_caching(self):
             ),
         )
 
+    @slow
+    @needs_wes
+    def test_wes_server_cwl_conformance(self):
+        """
+        Run the CWL conformance tests via WES. TOIL_WES_ENDPOINT must be
+        specified. If the WES server requires authentication, set TOIL_WES_USER
+        and TOIL_WES_PASSWORD.
+
+        To run manually:
+
+        TOIL_WES_ENDPOINT=http://localhost:8080 \
+        TOIL_WES_USER=test \
+        TOIL_WES_PASSWORD=password \
+        python -m pytest src/toil/test/cwl/cwlTest.py::CWLv12Test::test_wes_server_cwl_conformance -vv --log-level INFO --log-cli-level INFO
+        """
+        endpoint = os.environ.get("TOIL_WES_ENDPOINT")
+        user = os.environ.get("TOIL_WES_USER")
+        password = os.environ.get("TOIL_WES_PASSWORD")
+
+        extra_args = [f"--wes_endpoint={endpoint}"]
+        if user and password:
+            # TODO: if we run this on the CI, make sure we don't log this...
+            #  we output the entire command before we run, and cwltest logs the command if a test fails.
+            extra_args.extend([
+                f"--wes_user={user}",
+                f"--wes_password={password}"
+            ])
+
+        # These are the ones that currently fail:
+        #   - 310: mixed_version_v10_wf
+        #   - 311: mixed_version_v11_wf
+        #   - 312: mixed_version_v12_wf
+        #   - 331: iwd-fileobjs1
+        #   - 332: iwd-fileobjs2
+
+        # Main issues:
+        # 1. `cwltool --print-deps` doesn't seem to include secondary files from the default
+        #     e.g.: https://github.com/common-workflow-language/cwl-v1.2/blob/1.2.1_proposed/tests/mixed-versions/wf-v10.cwl#L4-L10
+        # 2. We need a way to reinterpret relative paths inside the workflow files
+        #    e.g.: https://github.com/common-workflow-language/cwl-v1.2/blob/1.2.1_proposed/tests/iwd/iwd-fileobjs1.cwl#L7-L11
+
+        return self.test_run_conformance(
+            runner=["python", os.path.abspath("src/toil/test/server/wes_cwl_runner.py")],
```

Comment:
Maybe add another entry point for this script to expose it as a command, like we have for `toil-cwl-runner`?